management-server/api/proto/wso2/discovery/api/api.proto

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
+syntax = "proto3";
+
+option go_package = "github.com/wso2/apk/management-server/wso2/discovery/api";
+package wso2.discovery.api;
+
+service APIService {
+    rpc createAPI(API) returns (Response);
+    rpc updateAPI(API) returns (Response);
+    rpc deleteAPI(API) returns (Response);
+}
+
+message API {
+    string uuid = 1;
+    string provider = 2;
+    string version = 3;
+    string name = 4;
+    string context = 5;
+    string type = 6;
+    string organizationId = 7;
+    string createdBy = 8;
+    string createdTime = 9;
+    string updatedBy = 10;
+    string updatedTime = 11;
+    string definition = 12;
+    repeated string transport = 13; 
```

Comment:
For repeated fields the naming convention is to use the plural form.
https://developers.google.com/protocol-buffers/docs/style#repeated_fields